homeassistant/components/zeversolar/config_flow.py

```diff
@@ -0,0 +1,61 @@
+"""Config flow for zeversolar integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+import zeversolar
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_IP_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS): cv.string,
+    },
+)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for zeversolar."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        client = zeversolar.ZeverSolarClient(host=user_input[CONF_IP_ADDRESS])
+        try:
+            data = await self.hass.async_add_executor_job(client.get_data)
+        except zeversolar.ZeverSolarHTTPNotFound:
+            errors["base"] = "invalid_host"
+        except zeversolar.ZeverSolarHTTPError:
+            errors["base"] = "cannot_connect"
+        except zeversolar.ZeverSolarTimeout:
+            errors["base"] = "timeout_connect"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            await self.async_set_unique_id(data.serial_number)
+            self._abort_if_unique_id_configured()
+            return self.async_create_entry(title=data.serial_number, data=user_input)
```

Comment:
A serial number isn't always the best title.
It might be better to use the integration title/name instead, especially since it is safe to assume one would generally use a single instance of this integration. The UI will, in that case, hide the title until the user assigns a custom name.

Reply:
```suggestion
            return self.async_create_entry(title="Zeversolar", data=user_input)
```